assets/src/design-system/components/dropDown/menu/index.js

```diff
@@ -0,0 +1,170 @@
+
+export const DropDownMenu = ({
+  anchorHeight,
+  dropDownHeight,
+  emptyText,
+  menuStylesOverride,
+  hasMenuRole,
+  isRTL,
+  options = [],
+  listId,
+  onMenuItemClick,
+  onDismissMenu,
+  renderItem = DefaultListItem,
+  activeValue,
+  menuAriaLabel,
+}) => {
+  const ListItem = renderItem;
+  const listRef = useRef();
+  const optionsRef = useRef([]);
+
+  const handleMenuItemSelect = useCallback(
+    (event, { value }) => onMenuItemClick(event, value),
+    [onMenuItemClick]
+  );
+
+  const { focusedIndex, listLength } = useDropDownMenu({
+    activeValue,
+    handleMenuItemSelect,
+    isRTL,
+    options,
+    listRef,
+    onDismissMenu,
+  });
+
+  useEffect(() => {
+    const listEl = listRef.current;
+    if (!listEl || focusedIndex === null) {
+      return;
+    }
+    if (focusedIndex === -1) {
+      listEl.scrollTo(0, 0);
+      return;
+    }
+
+    const highlighedOptionEl = optionsRef.current[focusedIndex];
+    if (!highlighedOptionEl) {
+      return;
+    }
+
+    highlighedOptionEl.focus();
+    listEl.scrollTo(0, highlighedOptionEl.offsetTop - listEl.clientHeight / 2);
+  }, [focusedIndex]);
+
+  const renderMenuLabel = (label) => {
+    return (
+      <GroupLabel id={`dropDownMenuLabel-${label}`} role="presentation">
+        {label}
+      </GroupLabel>
+    );
+  };
+
+  const renderMenuItem = useCallback(
+    (option, optionIndex, groupIndex = 0) => {
+      const isSelected = option.value === activeValue;
+      const optionInset = getInset(options, groupIndex, optionIndex);
+      return (
+        <ListItem
+          aria-posinset={optionInset}
+          aria-selected={isSelected}
+          aria-setsize={listLength}
+          id={`dropDownMenuItem-${option.value}`}
+          key={option.value}
+          role={hasMenuRole ? 'menuitem' : 'option'}
+          onClick={(event) => handleMenuItemSelect(event, option)}
+          tabIndex={0}
+          ref={(el) => (optionsRef.current[optionInset] = el)}
+          option={option}
+          isSelected={isSelected}
+        />
+      );
+    },
+    [activeValue, options, listLength, hasMenuRole, handleMenuItemSelect]
+  );
+
+  const MenuContent = useMemo(() => {
+    if (!options || options.length === 0) {
+      return <EmptyList>{emptyText}</EmptyList>;
+    }
+    const isManyGroups = options.length > 1;
+    return options.map(({ label, group }, groupIndex) => {
+      const groupAria = isManyGroups
+        ? { 'aria-label': label }
+        : { 'aria-labelledby': listId };
+
+      return (
+        <List
+          key={label || `menuGroup_${groupIndex}`}
+          role="group"
+          isNested={isManyGroups}
+          {...groupAria}
+        >
+          {label && renderMenuLabel(label)}
+          {group.map((groupOption, optionIndex) =>
+            renderMenuItem(groupOption, optionIndex, groupIndex)
+          )}
+        </List>
+      );
+    });
+  }, [options, emptyText, renderMenuItem, listId]);
```

Comment:
In this functional component we have `MenuContent`, `renderMenuItem`, `renderMenuLabel`. I think they are big and complex enough to become their own components. The latter two could turn into components like `<MenuItem />` and `<MenuLabel />` with props too. While they are memoized, they still need to be checked on each render. This should also make them more composable.

Reply:
Thanks for calling this out, I was waffling between this. Updating now :)

Reply:
Haah I was on the fence about updating this too. glad it got called out

Reply:
@carlos-kelly @littlemilkstudio thoughts about this update? 

I pulled apart anything that was rendering into it's own component and file. So `/index` is now more of a container that gives functionality to the list. If there's no options them the `EmptyList` component takes over. Otherwise, we have `ListGroupings` which then sets up a `List` with a `GroupLabel` if relevant per group and then proceeds to map through options, where if `renderItem` was passed into the `<DropDown />` it will override `ListItem` but otherwise will default to use `menu/defaultListItem`. 

I think this gives us more room to test edge cases in the menu itself and I think the naming makes it clear what's happening.

Reply:
I love it. Mush easier to read and reason about. Great work!